### ISSUE TITLE
Remote building with `--sandbox` doesn't produce a sandbox container

### ISSUE THREAD

**Issue by WestleyK** [apptainer/singularity#2415]
## Version of Singularity:

```
3.0.1-145.gc9822fec
```

### Expected behavior:

Build succeeds, and produce a sandbox container:

```
$ singularity build --remote --sandbox sand.sif docker://ubuntu:latest
[...]
$ ls
sand.sif  # a sandbox directory
```

<br>

### Actual behavior

Does not producing a sandbox container:

```
$ singularity build --remote --sandbox sand.sif docker://ubuntu:latest
Getting image source signatures
Copying blob sha256:32802c0cfa4defde2981bec336096350d0bb490469c494e21f678b1dcf6d831f
 30.62 MiB / 30.62 MiB  1s
Copying blob sha256:da1315cffa03c17988ae5c66f56d5f50517652a622afc1611a8bdd6c00b1fde3
 847 B / 847 B  0s
Copying blob sha256:fa83472a3562898caaf8d77542181a473a84039376f2ba56254619d9317ba00d
 556 B / 556 B  0s
Copying blob sha256:f85999a86bef2603a9e9a4fa488a7c1f82e471cbb76c3b5068e54e1a9320964a
 162 B / 162 B  0s
Copying config sha256:dec80b78b0e6d45cbe2e098c747d28dc7c4ca39cab2608fe22790e68098ae7c4
 2.42 KiB / 2.42 KiB  0s
Writing manifest to image destination
Storing signatures
INFO:    Creating SIF file...
INFO:    Build complete: /tmp/image-242422562
INFO:    Now uploading /tmp/image-242422562 to the library
 26.69 MiB / 26.69 MiB  100.00% 31.69 MiB/s 0s
INFO:    Setting tag latest
 26.69 MiB / 26.69 MiB [=========================================================================================================================================] 100.00% 3.27 MiB/s 8s
$ ls
sand.sif  # not a directory
```

<br>

### Steps to reproduce behavior

```
$ singularity build --remote --sandbox sand.sif docker://ubuntu:latest
```

**Comment by MilesCranmer** [comment]
@ikaneshiro et al - I'm not sure if this works yet. I see the following error when I run:
```
singularity build --remote --sandbox sand.sif docker://ubuntu:latest
```
Error:
```
...
INFO:    Creating SIF file...
INFO:    Build complete: /tmp/image-777972103
WARNING: Skipping container verifying
 26.48 MiB / 26.48 MiB  100.00% 53.50 MiB/s 0s
WARNING: 'nodev' mount option set on /tmp, it could be a source of failure during build process
INFO:    Starting build...
FATAL:   While performing build: conveyor failed to get: failed to retrieve path for /tmp/remote-build-610447928: lstat /tmp/remote-build-610447928: no such file or directory
```

I'm on singularity 3.5.2. The file actually appears, but when I execute it with "--writeable", I see:
```
FATAL:   no SIF writable overlay partition found in /.../sand.sif
```

Let me know if I should open this as a different issue.

**Comment by ikaneshiro** [comment]
Hi @MilesCranmer This appears to be a different issue with these options, so please open a new issue to track this and I'll start to work on a fix.

**Comment by ikaneshiro** [comment]
Hi again @MilesCranmer, I was able to replicate your build failure and have submitted a fix that you can test in #5568. 

I was not able to replicate your issue with `--writable`. If that is still an issue for you, please open an issue with some debug information for us to try to figure out the cause and create a fix. Thanks!

**Comment by MilesCranmer** [comment]
Thanks, and sorry for the late response! Indeed this looks to solve my issue. Cheers!
Miles
